src/providers/oauth-provider.js

```diff
@@ -107,23 +107,16 @@ function OAuthProvider() {
        * Retrieves the `access_token` and stores the `response.data` on cookies
        * using the `OAuthToken`.
        *
-       * @param {object} user - Object with `username` and `password` properties.
-       * @param {object} config - Optional configuration object sent to `POST`.
+       * @param {object} data - Request content, e.g., `username` and `password`.
+       * @param {object} options - Optional configuration.
        * @return {promise} A response promise.
        */
 
-      getAccessToken(user, options) {
-        // Check if `user` has required properties.
-        if (!user || !user.username || !user.password) {
```

Comment:
We now support data extension. So, we cannot assume that data will come with username/password.